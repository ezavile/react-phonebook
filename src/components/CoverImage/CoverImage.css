.CoverImage {
  position: relative;
  width: 100%;
  height: 18em;
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
}

.CoverImage-title {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 1em;
  background: rgba(0, 0, 0, 0.54);
  color: white;
}
