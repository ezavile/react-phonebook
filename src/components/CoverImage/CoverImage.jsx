import React from 'react';
import PropTypes from 'prop-types';
import './CoverImage.css';

const CoverImage = ({ image, title }) => (
  <div className="CoverImage" style={{ backgroundImage: `url(${image})` }}>
    <h1 className="CoverImage-title">{title}</h1>
  </div>
);

CoverImage.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default CoverImage;
