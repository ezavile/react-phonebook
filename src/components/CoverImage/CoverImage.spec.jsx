import React from 'react';
import { shallow } from 'enzyme';
import CoverImage from './CoverImage';

describe('CoverImageComponent', () => {
  let coverImageComponent;

  beforeEach(() => {
    coverImageComponent = shallow(
      <CoverImage image="/path/img.png" title="some title" />
    );
  });

  it('should render <CoverImage />', () => {
    expect(coverImageComponent.getElements()).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(coverImageComponent.find('h1').text()).toBe('some title');
  });

  it('should have an image as a style', () => {
    expect(coverImageComponent.find('.CoverImage').html()).toContain(
      'url(/path/img.png)'
    );
  });
});
