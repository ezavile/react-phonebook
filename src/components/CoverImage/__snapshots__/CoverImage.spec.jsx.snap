// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoverImageComponent should render <CoverImage /> 1`] = `
Array [
  <div
    className="CoverImage"
    style={
        Object {
            "backgroundImage": "url(/path/img.png)",
          }
    }
>
    <h1
        className="CoverImage-title"
    >
        some title
    </h1>
</div>,
]
`;
